Add webpack support

I'm able to bundle gl-js using webpack, but I'm getting a runtime error (in glify) with Chrome 46 on OSX.  @tmcw mentioned this has been reported before and that a webpack alternative to glify may be needed.  Adding an issue for reference and noting workaround.

```
Uncaught TypeError: Cannot read property 'match' of undefined
```
at https://github.com/mapbox/glify/blob/master/index.js#L13

Simple example showing my work -- https://github.com/twelch/react-mapbox-gl-js/tree/webpack-test
One workaround is to shim with the pre-built gl-js library using the webpack script-loader module.
```
require("script!mapbox-gl/dist/mapbox-gl.js");
```
instead of
```
import mapboxgl from 'mapbox-gl';
```

Tried everything suggested but still getting the **Cannot find module 'mapbox-gl'** error. Could anyone please help?

`import * as mapboxgl from 'mapbox-gl';`

webpack.config.js

```
var path = require('path');
var webpack = require('webpack');
var merge = require('extendify')({
    isDeep: true,
    arrays: 'concat'
});
var ExtractTextPlugin = require('extract-text-webpack-plugin');
var extractCSS = new ExtractTextPlugin('styles.css');
var devConfig = require('./webpack.config.dev');
var prodConfig = require('./webpack.config.prod');
var isDevelopment = process.env.ASPNET_ENV === 'Development';

module.exports = merge({
    resolve: {
        extensions: ['', '.js', '.ts'],
        alias: {
            webworkify: 'webworkify-webpack'
        }
    },
    module: {
        loaders: [{
            test: /\.ts$/,
            include: /ClientApp/,
            loader: 'ts-loader'
        }, {
            test: /\.html$/,
            loader: 'raw-loader'
        }, {
            test: /\.css/,
            loader: extractCSS.extract(['css'])
        }, {
            test: /\.json$/,
            loader: 'json-loader'
        }, {
            test: /\.js$/,
            include: path.resolve(__dirname, 'node_modules/mapbox-gl/js/render/painter/use_program.js'),
            loader: 'transform/cacheable?brfs'
        }, {
            test: /\.js$/,
            include: path.resolve(__dirname, 'node_modules/webworkify/index.js'),
            loader: 'worker'
        }],
        postLoaders: [{
            include: /node_modules\/mapbox-gl/,
            loader: 'transform',
            query: 'brfs'
        }]
    },
    entry: {
        main: ['./ClientApp/boot-client.ts']
    },
    output: {
        path: path.join(__dirname, 'wwwroot', 'dist'),
        filename: '[name].js',
        publicPath: '/dist/'
    },
    plugins: [
        extractCSS,
        new webpack.DllReferencePlugin({
            context: __dirname,
            manifest: require('./wwwroot/dist/vendor-manifest.json')
        })
    ]
}, isDevelopment ? devConfig : prodConfig);
```

--- a/README.md
+++ b/README.md
@@ -61,6 +61,34 @@ var map = new mapboxgl.Map({
 });
 ```
 
+## Using Mapbox GL JS with [Webpack](https://webpack.github.io/)
+
+To use the [vector tiles](https://www.mapbox.com/maps/) and styles hosted on http://mapbox.com, you must [create an account](https://www.mapbox.com/studio/signup/) and then [obtain an access token](https://www.mapbox.com/studio/account/tokens/). You may learn more about access tokens [here](https://www.mapbox.com/help/define-access-token/).
+
+Install the [`mapbox-gl` npm package](https://www.npmjs.com/package/mapbox-gl)
+and the required loaders.
+
+```bash
+npm install --save mapbox-gl
+npm install --save transform-loader
+npm install --save json-loader
+npm install --save webworkify-webpack
+```
+
+Add the [required additional options from webpack.config.example.js](webpack.config.example.js)
+to your webpack configuration.
+
+Instantiate `mapboxgl.Map`
+
+```js
+var mapboxgl = require('mapbox-gl');
+mapboxgl.accessToken = '<your access token here>';
+var map = new mapboxgl.Map({
+    container: '<your HTML element id>',
+    style: 'mapbox://styles/mapbox/streets-v8'
+});
+```
+
 ## Contributing to Mapbox GL JS
 
 See [CONTRIBUTING.md](https://github.com/mapbox/mapbox-gl-js/blob/master/CONTRIBUTING.md).

--- a/README.md
+++ b/README.md
@@ -89,6 +89,14 @@ var map = new mapboxgl.Map({
 });
 ```
 
+### Using import
+
+If you're using the ES6 module system, you can import `mapboxgl` like so:
+
+```js
+import mapboxgl from 'mapbox-gl';
+```
+
 ## Contributing to Mapbox GL JS
 
 See [CONTRIBUTING.md](https://github.com/mapbox/mapbox-gl-js/blob/master/CONTRIBUTING.md).

--- a/webpack.config.example.js
+++ b/webpack.config.example.js
@@ -1,0 +1,63 @@
+const webpack = require('webpack');
+const path = require('path');
+
+/**
+ * Mapbox GL JS example webpack configuration. This demonstrates the options
+ * you'll need to add to your existing webpack configuration in order to successfully
+ * use Mapbox GL JS.
+ *
+ * This configuration refers to specific paths, especially in its configuration
+ * of the brfs transform. You'll likely need to change these paths so that
+ * they point at the correct file in Mapbox GL JS.
+ *
+ * Additional dependencies:
+ *
+ *     npm install --save transform-loader
+ *     npm install --save json-loader
+ *     npm install --save webworkify-webpack
+ *
+ * Why these dependencies?
+ *
+ * json-loader
+ * By default, browserify and node allow you to use the `require` method
+ * with JSON files. Webpack doesn't have this built-in, so we include
+ * the json loader.
+ *
+ * transform-loader
+ * Mapbox GL JS uses the brfs browserify transform to allow it to use
+ * the fs.readFileSync method in order to load its shaders for WebGL. Adding
+ * the transform loader lets webpack use brfs as well.
+ *
+ * webworkify-webpack
+ * Mapbox GL JS uses the webworkify module by default, and that module
+ * does things very specific to browserify's module loading system. In this
+ * configuration, we alias webworkify to webworkify-webpack to add webpack
+ * support.
+ */
+module.exports = {
+    entry: './test.js',
+    output: {
+        path: './',
+        filename: 'test.bundle.js',
+    },
+    resolve: {
+        alias: {
+            'webworkify': 'webworkify-webpack'
+        }
+    },
+    module: {
+        loaders: [{
+            test: /\.json$/,
+            loader: 'json-loader'
+        }, {
+            test: /\.js$/,
+            include: path.resolve(__dirname, 'js/render/painter/use_program.js'),
+            loader: 'transform/cacheable?brfs'
+        }],
+        postLoaders: [{
+            include: /node_modules\/mapbox-gl/,
+            loader: 'transform',
+            query: 'brfs'
+        }]
+    }
+}

